extensions-core/druid-basic-security/src/main/java/org/apache/druid/security/basic/authentication/BasicHTTPAuthenticator.java

```diff
@@ -67,19 +70,49 @@ public BasicHTTPAuthenticator(
       @JsonProperty("initialInternalClientPassword") PasswordProvider initialInternalClientPassword,
       @JsonProperty("enableCacheNotifications") Boolean enableCacheNotifications,
       @JsonProperty("cacheNotificationTimeout") Long cacheNotificationTimeout,
-      @JsonProperty("credentialIterations") Integer credentialIterations
+      @JsonProperty("credentialIterations") Integer credentialIterations,
+      @JsonProperty("skipOnFailure") Boolean skipOnFailure,
+      @JsonProperty("credentialsValidator") CredentialsValidator credentialsValidator,
+      @JsonProperty("credentialsValidator.url") String url,
+      @JsonProperty("credentialsValidator.bindUser") String bindUser,
+      @JsonProperty("credentialsValidator.bindPassword") PasswordProvider bindPassword,
+      @JsonProperty("credentialsValidator.baseDn") String baseDn,
+      @JsonProperty("credentialsValidator.userSearch") String userSearch,
+      @JsonProperty("credentialsValidator.userAttribute") String userAttribute,
+      @JsonProperty("credentialsValidator.groupFilters") String[] groupFilters,
+      @JsonProperty("credentialsValidator.credentialVerifyDuration") Integer credentialVerifyDuration,
+      @JsonProperty("credentialsValidator.credentialMaxDuration") Integer credentialMaxDuration,
+      @JsonProperty("credentialsValidator.credentialCacheSize") Integer credentialCacheSize
```

Comment:
does above credentialsValidator.* ppts belong to credentialsValidator ? 
If that is the case, is it possible to move them inside credientialsValidator itself and access it from there ? or move the logic that uses these inside the validator ?

Reply:
Yes, and specific to ldap validator. The ldap specific one are referenced inside LDAPCredentialsValidator as well. So it seems redundant in BasicHTTPAuthenticator.java.
I'll look into cleaning it up a bit as you suggested, thanks

Reply:
Fixed, i've cleaned this up. credentialsValidator.* properties have been moved inside corresponding credentials validator